### DISCUSSION TITLE
Scene Duplicate Checker (run the phash generation task)

### DISCUSSION THREAD

**Discussion by David-Maisonave** [stashapp/stash#5415] [Q&A] [answered]
# How to run the phash generation task?

After I ran **Scene Duplicate Checker**, it produced a report that states the following in the top:
`Missing phashes for 225 scenes. Please run the phash generation task.`

How does a user run phash generation task? I haven't seen an option for this.

I have this as an option in my own plugin [DupFileManager](https://github.com/David-Maisonave/Axter-Stash/tree/main/plugins/DupFileManager), and currently that's the only way I know how to run Generate PHASH Matching task.

On related question, is Scene Duplicate Checker a plugin?
Can a plugin be create for Scene Tools or any Tools?

**Comment by feederbox826** [comment]
> How does a user run phash generation task?

Generate task, with the phash option enabled

**Comment by feederbox826** [comment]
> On related question, is Scene Duplicate Checker a plugin?
Can a plugin be create for Scene Tools or any Tools?

https://github.com/stashapp/stash/tree/develop/ui/v2.5/src/components/SceneDuplicateChecker doesn't seem like it but I believe there was discussions around it but not sure if anything came from it

**Comment by David-Maisonave** [comment]
> Generate task, with the phash option enabled

I can not find this option. I even did a keyword search for "phash" on my browser, and it only finds the DupFileManager task.
Under Generated Content, I have the following:
![GeneratedContent](https://github.com/user-attachments/assets/73e886ee-b0ea-4050-a5b6-5cc836285d1b)

By any chance are you referring to **Perceptual hashes**?

If not, can you please give me a screenshot of this option?

**Comment by David-Maisonave** [comment]
If **Perceptual hashes** is phash option, I like to create a new issue feature request to have this menu option changed to the following:
**Perceptual hashes (PHASH)**

**Comment by sleetx** [comment, marked as answer]
phash is perceptual hash. Make sure it's toggled on. You may or may not want the other options toggled on when you're ready to generate, it will do them in the same task.

**Comment by David-Maisonave** [comment]
> phash is perceptual hash.

Thanks for the clarification.
